schema/connector-discovery-schema.json

```diff
@@ -0,0 +1,370 @@
+        },
+        "label": {
+          "type": "string"
+        },
+        "format": {
+          "$ref": "#/definitions/connector_action_user_input_formats"
+        },
+        "select": {
+          "type": "object"
+        },
+        "min_length": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "max_length": {
+          "type": "integer",
+          "minimum": 0
+        }
+      },
+      "if": {
+        "properties": {
+          "format": {
+            "const": "select"
+          }
+        }
+      },
+      "then": {
+        "required": [
+          "select"
+        ]
+      },
+      "required": [
+        "id",
+        "label",
+        "format"
+      ]
+    },
+    "connector_action_request_params": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "env": {
+          "$ref": "#/definitions/input_env_variables"
+        }
+      },
+      "required": ["env"]
+    },
+    "configs": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/config"
+      }
+    },
+    "config": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "default": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string"
+        },
+        "description": {
+          "type": "object",
+          "additionalProperties": {
+            "not": {
+              "type": "object"
+            }
+          }
+        },
+        "validators": {
+          "type": "array",
+          "items": {"$ref": "#/definitions/config_validator"}
+        }
+      },
+      "required": ["description"]
+    },
+    "config_validator": {
+      "anyOf": [
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "required"
+            },
+            "description": {
+              "type": "object",
+              "additionalProperties": {
+                "not": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "required": ["type"]
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "regex"
+            },
+            "value": {
+              "type": "string"
+            },
+            "description": {
+              "type": "object",
+              "additionalProperties": {
+                "not": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "required": ["type", "value"]
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "max_length"
+            },
+            "value": {
+              "type": "integer"
```

Comment:
This was earlier "string" type.

Reply:
And I think it should remain string.  See discussion in ticket.